src/AlgAss/AlgCyc.jl

```diff
@@ -0,0 +1,483 @@
+
+See also [`is_split_with_map`](@ref).
+
+# Example
+
+```julia
+julia> QQx, x = QQ[:x];
+julia> k, g = number_field(x^2 + 1);
+julia> is_split(cyclic_algebra(k, hom(k, k, -g), QQ(4)))
+true
+```
+"""
+function is_split(
+  c::CyclicAlgebra{T, S},
+)::Bool where {T, S}
+  return first(is_norm(c.cyc_fld, c.a))
+end
+
+"""
+Determine whether the cyclic algebra splits, and compute an iso to the matrix algebra.
+
+If the algebra does not split, the null map is returned. This function is not pure.
+
+See also [`is_split`](@ref).
+
+# Example
+
+```julia
+julia> QQx, x = QQ[:x];
+julia> k, g = number_field(x^2 + 1);
+julia> c = cyclic_algebra(k, hom(k, k, -g), QQ(4));
+julia> _, f = is_split_with_map(c);
+julia> f(generating_element(c))
+[2 0; 0 -2]
+```
+"""
+function is_split_with_map(
+  c::CyclicAlgebra{T, S},
+)::Tuple{Bool, Map} where {T, S}
+  d = degree(c.cyc_fld)
+  g = gen(c.cyc_fld)
+  k = c.cyc_fld
+  mat_alg = matrix_algebra(base_field(k), d)
+  emb1 = hom(k, mat_alg, mat_alg(representation_matrix(g)); check=false)
+  emb2 = hom(k, mat_alg, mat_alg(representation_matrix(c.sigma(g))); check=false)
+  sn = skolem_noether_conjugator(emb2, emb1; check=false)
+  if first(local _, sol = is_norm(k, c.a / (sn^d)[1, 1]))
+    m_pi = emb1(sol)*sn
+    return true, hom(c.sca, mat_alg, collect(
+      emb1(x)*m_pi^j
+      for x in basis(k), j in 0:d - 1
+    )[:]; check=false)
+  end
+  return false, hom(c.sca, mat_alg, [mat_alg(0) for _ in 1:d^2]; check=false)
+end
+
+"""
+Determine whether the given cyclic algebras are isomorphic.
+"""
+function is_isomorphic(
+  c1::CyclicAlgebra{T, S},
+  c2::CyclicAlgebra{T, S},
+)::Bool where {T, S}
+  return first(is_isomorphic_with_map(c1, c2))
+end
+
+"""
+Determine an isomorphism between two cyclic algebras.
+
+Returns a tuple, the first element being `true` in case the algebras are isomorphic. In
+this case the second tuple element is an isomorphism `c1` to `c2`, otherwise it is the
+null map. This function is not pure.
+
+See also [`is_isomorphic`](@ref).
+
+# Examples
+
+```julia
+julia> QQx, x = QQ[:x];
+julia> k, g = number_field(x^2 - 5);
+julia> c1 = cyclic_algebra(k, hom(k, k, -g), QQ(5));
+julia> c2 = cyclic_algebra(k, hom(k, k, -g), QQ(5 * 3^2));
+julia> first(is_isomorphic_with_map(c1, c2))
+true
+```
+
+# References
+
+Algorithm as described in [han07](@cite).
+"""
+function is_isomorphic_with_map(
+  c1::CyclicAlgebra{T},
+  c2::CyclicAlgebra{T};
+  linearly_disjoint::Bool = false,
+)::Tuple{Bool, Map} where T
+  d = degree(c1.cyc_fld)
+  k, k1, k2 = parent(c1.a), c1.cyc_fld, c2.cyc_fld
+  g, g1, g2 = gen(k), gen(k1), gen(k2)
+  a1, a2 = c1.a, c2.a
+
+  if dim(c1.sca)!= dim(c2.sca)
+    return false, hom(c1.sca, c2.sca, [c2.sca(0) for _ in 1:d^2]; check=false)
+  end
+
+  # Case: Base fields are isomorphic
+  if !linearly_disjoint && first(local _, iso = is_isomorphic_with_map(k1, k2))
+    i = 1
+    while ((iso ∘ c1.sigma^i ∘ inv(iso))(g2) != c2.sigma(g2))
```

Comment:
```suggestion
    while ((iso ∘ c1.sigma^i ∘ inv(iso))(g2) != c2.sigma(g2))
```
Sorry, I can't type this symbol. And creating the composition is a bit of overkill for just evaluating at one element.